developers/weaviate/concepts/indexing/hnsw-indexes.mdx

```diff
@@ -0,0 +1,156 @@
+
+A lower value for `ef` makes the search faster, but it might compromise on accuracy.
+
+The balance is crucial in scenarios where either speed or accuracy is a priority. For instance, in applications where rapid responses are critical, a lower `ef` might be preferable, even at the expense of some accuracy. In analytical or research contexts where precision is paramount, a higher `ef` might be better, despite the increased query time.
+
+### Configure ef
+
+You can set `ef` explicitly or allow Weaviate to set it dynamically. When `ef` is configured dynamically, Weaviate optimizes the balance between speed and recall based on real-time query requirements. If your query patterns are variable, consider setting dynamic `ef`.
+
+To enable dynamic `ef`, set `ef: -1` in your [collection configuration](/developers/weaviate/configuration/indexing-vector/hnsw-indexes#configure-an-hnsw-index). When dynamic `ef` is enabled, Weaviate adjusts the size of the ANN list based on the query response limit. The calculation also takes into account the values of `dynamicEfMin`, `dynamicEfMax`, and `dynamicEfFactor`.
+
+### Configure dynamic ef
+
+The `ef` parameter controls the size of the ANN list at query time. You can configure a specific list size or else let Weaviate configure the list dynamically. If you choose dynamic `ef`, Weaviate provides several options to control the size of the ANN list.
+
+The length of the list is determined by the query response limit that you set in your query. Weaviate uses the query limit as an anchor and modifies the size of ANN list according to the values you set for the `dynamicEf` parameters.
+
+- `dynamicEfMin` sets a lower bound on the list length.
+- `dynamicEfMax` sets an upper bound on the list length.
+- `dynamicEfFactor` sets a range for the list.
+
+To keep search recall high, the actual dynamic `ef` value stays above `dynamicEfMin` even if the query limit is small enough to suggest a lower value.
+
+To keep search speed reasonable even when retrieving large result sets, the dynamic `ef` value is limited to `dynamicEfMax`. Weaviate doesn't exceed `dynamicEfMax` even if the query limit is large enough to suggest a higher value.
+
+If the query limit is higher than `dynamicEfMax`, `dynamicEfMax` does not have any effect. In this case, the dynamic `ef` value is equal to the query limit.
+
+To determine the length of the ANN list, Weaviate multiples the query limit by `dynamicEfFactor`. The list range is modified by `dynamicEfMin` and `dynamicEfMax`.
+
+### Dynamic ef example
+
+Consider this GraphQL query that sets a limit of 4.
+
+```graphql
+{
+  Get {
+    JeopardyQuestion(limit: 4) {
+      answer
+      question
+    }
+  }
+}
+```
+
+Imagine the collection has dynamic `ef` configured.
+
+```json
+  "vectorIndexConfig": {
+     "ef": -1,
+     "dynamicEfMin": 5
+     "dynamicEfMax": 25
+     "dynamicEfFactor": 10
+  }
+```
+
+The resulting search list has these characteristics.
```

Comment:
> search list

I see the doc uses a few different versions of this, like "nearest neighbor list", "ANN list" and "search list". I think this could be confusing. The typical term imo is "candidate list", as it gets updated throughout the search process, and the top n results are returned once the algorithm terminates.